tests/test_client/tests.py

```diff
@@ -85,6 +87,83 @@ def test_post(self):
         self.assertEqual(response.templates[0].name, 'POST Template')
         self.assertContains(response, 'Data received')
 
+    def test_json_post(self):
+        """ POST some JSON data to a view """
+        post_data = {
+            'value': 37
+        }
+        response = self.client.post('/json_view/', post_data, content_type=JSON_CONTENT)
+
+        # Check some response details
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.context['data'], 37)
+        self.assertEqual(response.templates[0].name, 'POST Template')
+        self.assertContains(response, 'Viewing POST page.')
+
+    def test_json_put(self):
+        """ PUT some JSON data to a view """
+        put_data = {
+            'value': 37
+        }
+        response = self.client.put('/json_view/', put_data, content_type=JSON_CONTENT)
+
+        # Check some response details
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.context['data'], 37)
+        self.assertEqual(response.templates[0].name, 'PUT Template')
+        self.assertContains(response, 'Viewing PUT page.')
+
+    def test_json_patch(self):
+        """ PATCH some JSON data to a view """
+        patch_data = {
+            'value': 37
+        }
+        response = self.client.patch('/json_view/', patch_data, content_type=JSON_CONTENT)
+
+        # Check some response details
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.context['data'], 37)
+        self.assertEqual(response.templates[0].name, 'PATCH Template')
+        self.assertContains(response, 'Viewing PATCH page.')
+
+    def test_json_delete(self):
+        """ DELETE some JSON data to a view """
+        delete_data = {
+            'value': 37
+        }
+        response = self.client.delete('/json_view/', delete_data, content_type=JSON_CONTENT)
+
+        # Check some response details
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.context['data'], 37)
+        self.assertEqual(response.templates[0].name, 'DELETE Template')
+        self.assertContains(response, 'Viewing DELETE page.')
+
+    def test_encoded_json_post(self):
```

Comment:
IMO you can combine this test with `test_json_post()` by using `subTest()`, e.g.:
```python
for post_data in ({'value': 37}, '{"value": 37}'):
    with self.subTest(post_data=post_data):
        response = self.client.post('/json_view/', post_data, content_type=JSON_CONTENT)
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.context['data'], 37)
        self.assertEqual(response.templates[0].name, 'POST Template')
        self.assertContains(response, 'Viewing POST page.')
```

Reply:
They're testing slightly different things though. So I feel that warrants separate tests.

I also find subtest confuses things if there is a failure.